objectivec/Protobuf/Protobuf.h

```diff
@@ -0,0 +1,42 @@
+// Protocol Buffers - Google's data interchange format
+// Copyright 2008 Google Inc.  All rights reserved.
+// https://developers.google.com/protocol-buffers/
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// This header is meant to only be used by the generated source, it should not
+// be included in code using protocol buffers.
+
+#import <Foundation/Foundation.h>
+
+//! Project version number for ProtocolBuffers_dynamic.
+FOUNDATION_EXPORT double ProtocolBuffers_dynamicVersionNumber;
```

Comment:
What is this?  What's the "dynamic" part?  All version info should likely not be apple's versioning tool but instead be tied to the versioning of the protobuf library.

Reply:
Dynamic part is removed, that was a mistake in the naming when I first put this together. This header file is needed for framework module support. If module support is removed this file can get removed. The version numbers in here are derived from the Info.plist file which is mandatory in a framework.

Reply:
What actually creates the symbols?  Looks like apple's versioning might be enabled?  Shouldn't it somehow be wired up to the releases of this library instead?  (git tags or something)?

Reply:
I need to double check to make sure this is right but I believe it come from here: 
https://github.com/mdyson/protobuf/blob/master/objectivec/Protobuf/Info.plist#L20
Which links back to the projfile here: https://github.com/mdyson/protobuf/blob/master/objectivec/ProtocolBuffers_iOS.xcodeproj/project.pbxproj#L1289

You are right, this should be linked up to the library release versions, I will add this in an upcoming commit.

Reply:
These are decl for exported symbols, so it won't link back to the info.plist unless some code loads it and sets the globals to the same value.

Reply:
Did you get a chance to sort out where these symbols come from?